### DISCUSSION TITLE
Native-Image PGO instrument problem.

### DISCUSSION THREAD

**Discussion by iEAmi** [oracle/graal#7170] [Q&A] [answered]
Hi.
I try to compile my micronaut based project (with maven) with ```--pgo-instrument``` but I got this error:
```Error: -H:-TrackNodeSourcePosition is incompatible with --pgo-instrument```

I added ```-H:+TrackNodeSourcePosition``` to buildArgs but I got the same error.

I realized when I added ```-O``` (b or 1 or 2 is not important) option this happend

How I can fix this ?

**Comment by alina-yur** [comment, marked as answer]
hi @iEAmi. When building with PGO, the user can’t modify optimization level, because PGO implies optimizing for the best performance, and reducing opitimization level would conflict with that goal.
